File: tests/test-google.js
const { Builder, By, Key, until } = require('selenium-webdriver');

async function testGoogleSearch() {
  // Set up Chrome browser
  let driver = await new Builder().forBrowser('chrome').build();

  try {
    // Open Google
    await driver.get('https://www.google.com');

    // Find the search box, type in query, and press Enter
    let searchBox = await driver.findElement(By.name('q'));
    await searchBox.sendKeys('Selenium WebDriver JavaScript', Key.RETURN);

    // Wait until the title changes
    await driver.wait(until.titleContains('Selenium WebDriver JavaScript'), 20000);

    console.log('Test Passed: Title contains the search term');
  } catch (err) {
    console.error('Test Failed:', err);
  } finally {
    await driver.quit();
  }
}

testGoogleSearch();

File: tests/test_google.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

# Option 1: If chromedriver is in PATH
driver = webdriver.Chrome()

# Option 2: If you want to specify the path manually
# driver = webdriver.Chrome(executable_path="C:\\WebDrivers\\chromedriver.exe")

driver.get("https://www.google.com")
time.sleep(2)

search_box = driver.find_element(By.NAME, "q")
search_box.send_keys("Selenium WebDriver")
search_box.submit()

time.sleep(5)
driver.quit()
